bot/cogs/snakes.py

```diff
@@ -15,31 +38,67 @@ class Snakes:
+            return encode_url(name)
+
+        # Get a list of similar names if a match wasn't found
+        matches = difflib.get_close_matches(name, SNAKES, n=1)
+        return encode_url(matches[0] if matches != [] else random.choice(SNAKES))
+
+    @staticmethod
+    async def fetch(session, url):
+        """Fetch the contents of a URL as a json"""
+        async with async_timeout.timeout(10):
+            async with session.get(url) as response:
+                return await response.json()
+
     async def get_snek(self, name: str = None) -> Dict[str, Any]:
-        """
-        Go online and fetch information about a snake
+        """Get a snake with a given name, or otherwise randomly"""
+        if name is None:
+            name = random.choice([x.strip('\n') for x in SNAKES])
+            
+        elif name.upper() == "PYTHON":
+            return PYTHON
 
-        The information includes the name of the snake, a picture of the snake, and various other pieces of info.
-        What information you get for the snake is up to you. Be creative!
+        # Get snake information
+        async with aiohttp.ClientSession() as session:
+            url = self.snake_url(name)
 
-        If "python" is given as the snake name, you should return information about the programming language, but with
-        all the information you'd provide for a real snake. Try to have some fun with this!
+            # Get the 
+            response = await self.fetch(session, url)
+            page = response["query"]["pages"]
+            content = next(iter(page.values()))
 
-        :param name: Optional, the name of the snake to get information for - omit for a random snake
-        :return: A dict containing information on a snake
-        """
+            # Parse the full-res image from the thumbnail
+            thumb = content.get("thumbnail", {}).get("source", "http://i.imgur.com/HtIPyLy.png/beep")
+            image = "/".join(thumb.replace("thumb/", "").split("/")[:-1])
+            
+            # WHY WOULD YOU USE DICTIONARY LITERAL IN A RETURN STATEMENT but okay lol
+            return {
+                "name": content["title"],
+                "info": content.get("extract", "") or "I don't know about that snake!",
+                "image": image
+            }
 
     @command()
     async def get(self, ctx: Context, name: str = None):
-        """
-        Go online and fetch information about a snake
-
-        This should make use of your `get_snek` method, using it to get information about a snake. This information
-        should be sent back to Discord in an embed.
+        content = await self.get_snek(name)
+        # Just a temporary thing to make sure it's working
+        em = Embed()
+        em.title = content["name"]
+        em.description = content["info"][:1970] + "\n\nPS. If the image is a fucking map, blame wikipedia. -Somejuan"
+        em.set_image(url=content["image"])
```

Comment:
some of this can be more concise:
```py
embed=discord.Embed(
    title=content["name"],
    description=content["info"][:1970] + "\n\nPS. If the image is a fucking map, blame wikipedia. -Somejuan"
)
```